job-service/src/main/java/com/hpe/caf/services/job/api/filter/JobFilterQuery.java

```diff
@@ -0,0 +1,147 @@
+        JOB_NAME = JOB_TABLE.addColumn("name");
+        JOB_PARTITION_ID = JOB_TABLE.addColumn("partition_id");
+        JOB_CREATE_TIME = JOB_TABLE.addColumn("create_date");
+        JOB_LAST_MODIFIED = JOB_TABLE.addColumn("last_update_date");
+        JOB_STATUS = JOB_TABLE.addColumn("status");
+        JOB_PERCENTAGE_COMPLETE = JOB_TABLE.addColumn("percentage_complete");
+
+        LABEL_PARTITION_ID = LABEL_TABLE.addColumn("partition_id");
+        LABEL_JOB_ID = LABEL_TABLE.addColumn("job_id");
+        LABEL_LABEL = LABEL_TABLE.addColumn("label");
+        LABEL_VALUE = LABEL_TABLE.addColumn("value");
+
+        COLUMN_MAPPINGS = new HashMap<>();
+        COLUMN_MAPPINGS.put("id", JOB_ID);
+        COLUMN_MAPPINGS.put("name", JOB_NAME);
+        COLUMN_MAPPINGS.put("createTime", JOB_CREATE_TIME);
+        COLUMN_MAPPINGS.put("lastUpdateTime", JOB_LAST_MODIFIED);
+        COLUMN_MAPPINGS.put("status", JOB_STATUS);
+        COLUMN_MAPPINGS.put("percentageComplete", JOB_PERCENTAGE_COMPLETE);
+    }
+
+    public static Condition getQueryStatement(final String key, final ComparisonOperator comparisonOperator, final List<String> args)
+    {
+        if (key.startsWith("labels.")) {
+            final String[] keys = key.split("\\.");
+            final Condition con1 = BinaryCondition.equalTo(JOB_PARTITION_ID, LABEL_PARTITION_ID);
+            final Condition con2 = BinaryCondition.equalTo(JOB_ID, LABEL_JOB_ID);
+            final Condition con3 = BinaryCondition.equalTo(LABEL_LABEL, keys[1]);
+            final Condition con4 = convertConditionString(LABEL_VALUE, comparisonOperator, args);
+            return labelExistsCon(con1, con2, con3, con4);
+        } else {
+            return convertConditionString(COLUMN_MAPPINGS.get(key), comparisonOperator, args);
+        }
+    }
+
+    private static Condition convertConditionString(final DbColumn key, final ComparisonOperator comparisonOperator,
+                                                    final List<String> args)
+    {
+
+        final Condition condition;
+        final String value = args.get(0);
+        switch (comparisonOperator.getSymbol()) {
+            case "==":
+                condition = value.contains("*")
+                    ? BinaryCondition.like(key, value.replace("*", "%"))
+                    : BinaryCondition.equalTo(key, value);
+                break;
+            case "!=":
+                condition = value.contains("*")
+                    ? BinaryCondition.notLike(key, value.replace("*", "%"))
+                    : BinaryCondition.notEqualTo(key, value);
+                break;
+            case "=gt=":
+                condition = BinaryCondition.greaterThan(key, value);
+                break;
+            case "=ge=":
+                condition = BinaryCondition.greaterThanOrEq(key, value);
+                break;
+            case "=lt=":
+                condition = BinaryCondition.lessThan(key, value);
+                break;

```

Comment:
Setting `condition` is OK but I don't really see why you don't just return directly from the switch statement.

```suggestion
                return BinaryCondition.lessThan(key, value);
```